packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2824,4 +2826,39 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         this.state.isDeclareContext = oldIsDeclareContext;
       }
     }
+
+    parseMethod<T: N.MethodLike>(
+      node: T,
+      isGenerator: boolean,
+      isAsync: boolean,
+      isConstructor: boolean,
+      allowDirectSuper: boolean,
+      type: string,
+      inClassScope: boolean = false,
```

Comment:
nit: We can just use `...args` if none of them are used in the post processing logic. So we would not forget to sync signatures when we add even more arguments to `parseMethod`.

Reply:
Use of `...args` is one of the slowest alternatives. Just mention it in case perf is relevant here

Reply:
There are already some `...args` in our codebase. I don't know if this is a performance bottleneck, but at least I don't think we need to worry in this PR.

Reply:
In general. Everything that is a perf regression should be improved, but this is just a TS plugin so maybe no point because too much perf issues in this plugin.